### ISSUE TITLE
Use RFC 2119 keywords consistently

### ISSUE THREAD

**Issue by turt2live** [matrix-org/matrix-spec#1782]
https://tools.ietf.org/html/rfc2119

See https://github.com/matrix-org/matrix-spec/issues/701
See https://github.com/matrix-org/matrix-spec/issues/682

**Comment by clokep** [comment]
I'm in favor of standardizing this. It offloads any terminology discussion to the IETF and matches what people generally expect when reading specifications.

**Comment by turt2live** [comment]
ftr, as this affects the "presentation layer" of the spec, I don't believe this requires an MSC. It's something we can do without actually changing the meaning of the specification.

**Comment by Johennes** [comment]
Shall MSCs attempt to use the RFC2119 language as well or should that be reserved to the spec itself?

**Comment by richvdh** [comment]
You've all said the same thing as me, but for posterity here's the GH comment I started writing 6 hours ago:

---

I think we've said that use of RFC2119 language is to be encouraged in MSCs, especially as an MSC gets closer to FCP.

In the past we've tended to be more relaxed about wording in MSCs "as long as the intention is clear" but that has had some unfortunate consequences.

A well-written MSC often has a bunch of informative sections around a normative proposal. Obviously, only the normative section needs formal RFC2119 keywords.

In other words: MSCs SHOULD express the proposal using RFC2119 keywords.

**Comment by Johennes** [comment]
For reference: In https://github.com/matrix-org/matrix-spec/pull/1858#discussion_r1634728239 we discussed that the RFC keywords probably shouldn't be used in info or warning boxes because these usually should be descriptive rather than normative.

For what it's worth, this also matches the rules currently being used in gematik's extension of the spec. Below is an example of a normative requirement (delimited by `A_...` and `<=`) that uses keyword capitalization and an adjacent explanatory note ("Hinweis") that doesn't.

![Screenshot 2024-06-25 at 08 18 01](https://github.com/matrix-org/matrix-spec/assets/1137962/31206c32-8e1c-4480-818e-75512739f759)

**Comment by HarHarLinks** [comment]
> ftr, as this affects the "presentation layer" of the spec, I don't believe this requires an MSC. It's something we can do without actually changing the meaning of the specification.

While I agree with the comments made, I can not find that the spec defines its use of RFC 2119 language at all so far? If this is correct, I think that we would also need to do that, and I wonder if that change could affect the meaning of the spec somewhere.

**Comment by turt2live** [comment]
to clarify: the spec uses RFC 2119, but doesn't document that. PRs are very much welcome for adding that block of text.

This issue is to ensure the keywords are used where they're supposed to already be used. The history is a bit unclear on this front, but that's the state of affairs :D
